application/src/test/java/org/opentripplanner/updater/GtfsRealtimeFuzzyTripMatcherTest.java

```diff
@@ -2,47 +2,131 @@
+import org.opentripplanner.transit.service.TimetableRepository;
 
-public class GtfsRealtimeFuzzyTripMatcherTest extends GtfsTest {
+public class GtfsRealtimeFuzzyTripMatcherTest {
 
-  @Test
-  public void testMatch() {
-    String feedId = timetableRepository.getFeedIds().iterator().next();
+  private static final String ROUTE_ID = "r1";
+  private static final String FEED_ID = TimetableRepositoryForTest.FEED_ID;
+  private static final LocalDate SERVICE_DATE = LocalDate.of(2024, 11, 13);
+  private static final String GTFS_SERVICE_DATE = SERVICE_DATE.toString().replaceAll("-", "");
+  private static final int SERVICE_CODE = 555;
+  private static final SiteRepositoryBuilder siteRepositoryBuilder = SiteRepository.of();
+  private static final TimetableRepositoryForTest TEST_MODEL = new TimetableRepositoryForTest(
+    siteRepositoryBuilder
+  );
+  private static final RegularStop STOP_1 = TEST_MODEL.stop("s1").build();
+  private static final RegularStop STOP_2 = TEST_MODEL.stop("s2").build();
+  private static final TimetableRepository TIMETABLE_REPOSITORY = new TimetableRepository(
+    siteRepositoryBuilder.build(),
+    new Deduplicator()
+  );
+  private static final Route ROUTE = TimetableRepositoryForTest.route(id(ROUTE_ID)).build();
+  private static final String TRIP_ID = "t1";
+  private static final Trip TRIP = TimetableRepositoryForTest.trip(TRIP_ID).build();
+
+  private static final FeedScopedId SERVICE_ID = TimetableRepositoryForTest.id("sid1");
+  private static final String START_TIME = "07:30:00";
+  private static final RealTimeTripTimes TRIP_TIMES = TripTimesFactory.tripTimes(
+    TRIP,
+    TEST_MODEL.stopTimesEvery5Minutes(5, TRIP, START_TIME),
+    new Deduplicator()
+  );
+  private static final TripPattern TRIP_PATTERN = TimetableRepositoryForTest
+    .tripPattern("tp1", ROUTE)
+    .withStopPattern(TimetableRepositoryForTest.stopPattern(STOP_1, STOP_2))
+    .withScheduledTimeTableBuilder(builder -> builder.addTripTimes(TRIP_TIMES))
+    .build();
 
-    GtfsRealtimeFuzzyTripMatcher matcher = new GtfsRealtimeFuzzyTripMatcher(
-      new DefaultTransitService(timetableRepository)
+  @BeforeAll
+  static void setup() {
+    TRIP_TIMES.setServiceCode(SERVICE_CODE);
+    CalendarServiceData calendarServiceData = new CalendarServiceData();
+    calendarServiceData.putServiceDatesForServiceId(SERVICE_ID, List.of(SERVICE_DATE));
+    TIMETABLE_REPOSITORY.addTripPattern(TRIP_PATTERN.getId(), TRIP_PATTERN);
+    TIMETABLE_REPOSITORY.getServiceCodes().put(SERVICE_ID, SERVICE_CODE);
+    TIMETABLE_REPOSITORY.updateCalendarServiceData(
+      true,
+      calendarServiceData,
+      DataImportIssueStore.NOOP
     );
-    TripDescriptor trip1 = TripDescriptor
+    TIMETABLE_REPOSITORY.index();
+  }
+
+  @Test
+  void noTripId() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().build();
+    assertEquals(TRIP.getId().getId(), matcher.match(FEED_ID, trip1).getTripId());
```

Comment:
Why use TRIP.getId().getId() here instead of TRIP_ID like elsewhere?

Reply:
Thomas has requested it here: https://github.com/opentripplanner/OpenTripPlanner/pull/6250#discussion_r1853854631

Reply:
Fixed in f40626194bfc67e2bfdac64f41eb2d69d881d397